ipython doesn't start when `c.TerminalInteractiveShell.highlighting_style` is set to a `pygments` theme string

Potentially related to #14810.

To reproduce, create a fresh ipython config, (`ipython profile create`), then uncomment `c.TerminalInteractiveShell.highlighting_style` and set it to `fruity` (or any pygments theme, or any string at all, really). Result:

```sh
***************************************************************************

IPython post-mortem report

{'commit_hash': '9970f5e4b',
 'commit_source': 'installation',
 'default_encoding': 'utf-8',
 'ipython_path': '/Users/rben01/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython',
 'ipython_version': '9.0.2',
 'os_name': 'posix',
 'platform': 'macOS-15.3.1-arm64-arm-64bit',
 'sys_executable': '/Users/rben01/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/bin/python',
 'sys_platform': 'darwin',
 'sys_version': '3.12.9 (main, Feb 12 2025, 15:09:19) [Clang 19.1.6 ]'}

***************************************************************************

Application name: ipython

Current user configuration structure:

{'BaseIPythonApplication': {'verbose_crash': True},
 'TerminalInteractiveShell': {'highlighting_style': 'fruity'}}

***************************************************************************

Crash traceback:

---------------------------------------------------------------------------
---------------------------------------------------------------------------
AssertionErrorPython 3.12.9: /Users/rben01/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/bin/python
                                                   Mon Mar 10 00:31:06 2025
A problem occurred executing Python code.  Here is the sequence of function
calls leading up to the error, with the most recent (innermost) call last.
File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/bin/ipython:12
      1 #!/bin/sh
      2 '''exec' "$(dirname -- "$(realpath -- "$0")")"/'python' "$0" "$@"
      3 ' '''
      4 # -*- coding: utf-8 -*-
      5 import sys
      6 from IPython import start_ipython
      7 if __name__ == "__main__":
      8     if sys.argv[0].endswith("-script.pyw"):
      9         sys.argv[0] = sys.argv[0][:-11]
     10     elif sys.argv[0].endswith(".exe"):
     11         sys.argv[0] = sys.argv[0][:-4]
---> 12     sys.exit(start_ipython())

File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython/__init__.py:139, in start_ipython(argv=None, **kwargs={})
    113 def start_ipython(argv=None, **kwargs):
    114     """Launch a normal IPython instance (as opposed to embedded)
    115 
    116     `IPython.embed()` puts a shell in a particular calling scope,
   (...)    136         allowing configuration of the instance (see :ref:`terminal_options`).
    137     """
    138     from IPython.terminal.ipapp import launch_new_instance
--> 139     return launch_new_instance(argv=argv, **kwargs)
        launch_new_instance = <bound method Application.launch_instance of <class 'IPython.terminal.ipapp.TerminalIPythonApp'>>
        argv = None
        kwargs = {}
File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/traitlets/config/application.py:1074, in Application.launch_instance(cls=<class 'IPython.terminal.ipapp.TerminalIPythonApp'>, argv=None, **kwargs={})
   1067 @classmethod
   1068 def launch_instance(cls, argv: ArgvType = None, **kwargs: t.Any) -> None:
   1069     """Launch a global instance of this Application
   1070 
   1071     If a global instance already exists, this reinitializes and starts it
   1072     """
   1073     app = cls.instance(**kwargs)
-> 1074     app.initialize(argv)
        app = <IPython.terminal.ipapp.TerminalIPythonApp object at 0x102448d10>
        argv = None   1075     app.start()

File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/traitlets/config/application.py:118, in catch_config_error.<locals>.inner(app=<IPython.terminal.ipapp.TerminalIPythonApp object>, *args=(None,), **kwargs={})
    115 @functools.wraps(method)
    116 def inner(app: Application, *args: t.Any, **kwargs: t.Any) -> t.Any:
    117     try:
--> 118         return method(app, *args, **kwargs)
        method = <function TerminalIPythonApp.initialize at 0x104caaac0>
        app = <IPython.terminal.ipapp.TerminalIPythonApp object at 0x102448d10>
        args = (None,)
        kwargs = {}    119     except (TraitError, ArgumentError) as e:
    120         app.log.fatal("Bad config encountered during initialization: %s", e)
    121         app.log.debug("Config at the time: %s", app.config)
    122         app.exit(1)

File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython/terminal/ipapp.py:286, in TerminalIPythonApp.initialize(self=<IPython.terminal.ipapp.TerminalIPythonApp object>, argv=None)
    274 @catch_config_error
    275 def initialize(self, argv=None):
    276     """Do actions after construct, but before starting the app."""
    277     super(TerminalIPythonApp, self).initialize(argv)
    278     if self.subapp is not None:
    279         # don't bother initializing further, starting subapp
    280         return
    281     # print(self.extra_args)
    282     if self.extra_args and not self.something_to_run:
    283         self.file_to_run = self.extra_args[0]
    284     self.init_path()
    285     # create the shell
--> 286     self.init_shell()
        self = <IPython.terminal.ipapp.TerminalIPythonApp object at 0x102448d10>    287     # and draw the banner
    288     self.init_banner()
    289     # Now a variety of things that happen after the banner is printed.
    290     self.init_gui_pylab()
    291     self.init_extensions()
    292     self.init_code()

File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython/terminal/ipapp.py:300, in TerminalIPythonApp.init_shell(self=<IPython.terminal.ipapp.TerminalIPythonApp object>)
    294 def init_shell(self):
    295     """initialize the InteractiveShell instance"""
    296     # Create an InteractiveShell instance.
    297     # shell.display_banner should always be False for the terminal
    298     # based app, because we call shell.show_banner() by hand below
    299     # so the banner shows *before* all extension loading stuff.
--> 300     self.shell = self.interactive_shell_class.instance(parent=self,
        self = <IPython.terminal.ipapp.TerminalIPythonApp object at 0x102448d10>    301                     profile_dir=self.profile_dir,
    302                     ipython_dir=self.ipython_dir, user_ns=self.user_ns)
    303     self.shell.configurables.append(self)

File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/traitlets/config/configurable.py:583, in SingletonConfigurable.instance(cls=<class 'IPython.terminal.interactiveshell.TerminalInteractiveShell'>, *args=(), **kwargs={'ipython_dir': '/Users/rben01/.ipython', 'parent': <IPython.terminal.ipapp.TerminalIPythonApp object>, 'profile_dir': <IPython.core.profiledir.ProfileDir object>, 'user_ns': None})
    553 @classmethod
    554 def instance(cls: type[CT], *args: t.Any, **kwargs: t.Any) -> CT:
    555     """Returns a global instance of this class.
    556 
    557     This method create a new instance if none have previously been created
   (...)    579         True
    580     """
    581     # Create and save the instance
    582     if cls._instance is None:
--> 583         inst = cls(*args, **kwargs)
        cls = <class 'IPython.terminal.interactiveshell.TerminalInteractiveShell'>
        args = ()
        kwargs = {'parent': <IPython.terminal.ipapp.TerminalIPythonApp object at 0x102448d10>, 'profile_dir': <IPython.core.profiledir.ProfileDir object at 0x104c974d0>, 'ipython_dir': '/Users/rben01/.ipython', 'user_ns': None}    584         # Now make sure that the instance will also be returned by
    585         # parent classes' _instance attribute.
    586         for subclass in cls._walk_mro():
    587             subclass._instance = inst
    589     if isinstance(cls._instance, cls):
    590         return cls._instance
    591     else:
    592         raise MultipleInstanceError(
    593             f"An incompatible sibling of '{cls.__name__}' is already instantiated"
    594             f" as singleton: {type(cls._instance).__name__}"
    595         )

File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython/terminal/interactiveshell.py:977, in TerminalInteractiveShell.__init__(self=<IPython.terminal.interactiveshell.TerminalInteractiveShell object>, *args=(), **kwargs={'ipython_dir': '/Users/rben01/.ipython', 'parent': <IPython.terminal.ipapp.TerminalIPythonApp object>, 'profile_dir': <IPython.core.profiledir.ProfileDir object>, 'user_ns': None})
    976 def __init__(self, *args, **kwargs) -> None:
--> 977     super(TerminalInteractiveShell, self).__init__(*args, **kwargs)
        self = <IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x104c30f80>
        args = ()
        kwargs = {'parent': <IPython.terminal.ipapp.TerminalIPythonApp object at 0x102448d10>, 'profile_dir': <IPython.core.profiledir.ProfileDir object at 0x104c974d0>, 'ipython_dir': '/Users/rben01/.ipython', 'user_ns': None}
        TerminalInteractiveShell = <class 'IPython.terminal.interactiveshell.TerminalInteractiveShell'>    978     self._set_autosuggestions(self.autosuggestions_provider)
    979     self.init_prompt_toolkit_cli()
    980     self.init_term_title()
    981     self.keep_running = True
    982     self._set_formatter(self.autoformatter)

File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython/core/interactiveshell.py:628, in InteractiveShell.__init__(self=<IPython.terminal.interactiveshell.TerminalInteractiveShell object>, ipython_dir='/Users/rben01/.ipython', profile_dir=<IPython.core.profiledir.ProfileDir object>, user_module=None, user_ns=None, custom_exceptions=((), None), **kwargs={'parent': <IPython.terminal.ipapp.TerminalIPythonApp object>})
    601 self.init_profile_dir(profile_dir)
    602 self.init_instance_attrs()
    603 self.init_environment()
    605 # Check if we're in a virtualenv, and set up sys.path.
    606 self.init_virtualenv()
    608 # Create namespaces (user_ns, user_global_ns, etc.)
    609 self.init_create_namespaces(user_module, user_ns)
    610 # This has to be done after init_create_namespaces because it uses
    611 # something in self.user_ns, but before init_sys_modules, which
    612 # is the first thing to modify sys.
    613 # TODO: When we override sys.stdout and sys.stderr before this class
    614 # is created, we are saving the overridden ones here. Not sure if this
    615 # is what we want to do.
    616 self.save_sys_module_state()
    617 self.init_sys_modules()
    619 # While we're trying to have each part of the code directly access what
    620 # it needs without keeping redundant references to objects, we have too
    621 # much legacy code that expects ip.db to exist.
    622 self.db = PickleShareDB(os.path.join(self.profile_dir.location, 'db'))
    624 self.init_history()
    625 self.init_encoding()
    626 self.init_prefilter()
--> 628 self.init_syntax_highlighting()
        self = <IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x104c30f80>    629 self.init_hooks()
    630 self.init_events()
    631 self.init_pushd_popd_magic()
    632 self.init_user_ns()
    633 self.init_logger()
    634 self.init_builtins()
    636 # The following was in post_config_initialization
    637 self.raw_input_original = input
    638 self.init_completer()
    639 # TODO: init_io() needs to happen before init_traceback handlers
    640 # because the traceback handlers hardcode the stdout/stderr streams.
    641 # This logic in in debugger.Pdb and should eventually be changed.
    642 self.init_io()
    643 self.init_traceback_handlers(custom_exceptions)
    644 self.init_prompts()
    645 self.init_display_formatter()
    646 self.init_display_pub()

File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython/core/interactiveshell.py:799, in InteractiveShell.init_syntax_highlighting(self=<IPython.terminal.interactiveshell.TerminalInteractiveShell object>, changes=None)
    774 @observe("colors")
    775 def init_syntax_highlighting(self, changes=None):
    776     # Python source parser/formatter for syntax highlighting
    777     pyformat = PyColorize.Parser(theme_name=self.colors).format
    778     self.pycolorize = lambda src: pyformat(src, "str")
    779     if not hasattr(self, "inspector"):
    780         self.inspector = self.inspector_class(
    781             theme_name=self.colors,
    782             str_detail_level=self.object_info_string_level,
    783             parent=self,
    784         )
    786     try:
    787         # Deprecation in 9.0, colors should always be lower
    788         self.inspector.set_theme_name(self.colors.lower())
    789     except Exception:
    790         warn(
    791             "Error changing object inspector color schemes.\n%s"
    792             % (sys.exc_info()[1]),
    793             stacklevel=2,
    794         )
    795     if hasattr(self, "InteractiveTB"):
    796         self.InteractiveTB.set_theme_name(self.colors)
    797     if hasattr(self, "SyntaxTB"):
    798         self.SyntaxTB.set_theme_name(self.colors)
--> 799     self.refresh_style()
        self = <IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x104c30f80>
File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython/terminal/interactiveshell.py:358, in TerminalInteractiveShell.refresh_style(self=<IPython.terminal.interactiveshell.TerminalInteractiveShell object>)
    357 def refresh_style(self):
--> 358     self._style = self._make_style_from_name_or_cls(self.highlighting_style)
        self = <IPython.terminal.interactiveshell.TerminalInteractiveShell object at 0x104c30f80>
File ~/.cache/uv/archive-v0/RUiK-9Ks_FFjrLHQhIAD-/lib/python3.12/site-packages/IPython/terminal/interactiveshell.py:822, in TerminalInteractiveShell._make_style_from_name_or_cls(self=<IPython.terminal.interactiveshell.TerminalInteractiveShell object>, name_or_cls='fruity')
    816 def _make_style_from_name_or_cls(self, name_or_cls):
    817     """
    818     Small wrapper that make an IPython compatible style from a style name
    819 
    820     We need that to add style for prompt ... etc.
    821     """
--> 822     assert name_or_cls == "legacy"
        name_or_cls == "legacy" = False
        name_or_cls = 'fruity'    823     legacy = self.colors.lower()
    825     theme = theme_table.get(legacy, None)
    826     assert theme is not None, legacy
    828     if legacy == "nocolor":
    829         style_overrides = {}
    830         style_cls = _NoStyle
    831     else:
    832         style_overrides = {**theme.extra_style, **self.highlighting_style_overrides}
    833         if theme.base is not None:
    834             style_cls = get_style_by_name(theme.base)
    835         else:
    836             style_cls = _NoStyle
    838     style = merge_styles(
    839         [
    840             style_from_pygments_cls(style_cls),
    841             style_from_pygments_dict(style_overrides),
    842         ]
    843     )
    845     return style

AssertionError: 

***************************************************************************

History of session input:
```

--- a/IPython/terminal/interactiveshell.py
+++ b/IPython/terminal/interactiveshell.py
@@ -355,7 +355,7 @@ class TerminalInteractiveShell(InteractiveShell):
             return
 
     def refresh_style(self):
-        self._style = self._make_style_from_name_or_cls(self.highlighting_style)
+        self._style = self._make_style_from_name_or_cls("legacy")
 
     # TODO: deprecate this
     highlighting_style_overrides = Dict(

